bin/genome-env-next

```diff
@@ -0,0 +1,432 @@
+                ;;
+            s)
+                SNAPSHOT_NAME="$OPTARG"
+                ;;
+            U)
+                DISABLE_UR="TRUE"
+                ;;
+            u)
+                UR_REPO="$OPTARG"
+                ;;
+            v)
+                VERBOSE="TRUE"
+                ;;
+            W)
+                DISABLE_WF="TRUE"
+                ;;
+            w)
+                WF_REPO="$OPTARG"
+                ;;
+            :)
+                err "option -$OPTARG requires an argument"
+                exit 1
+                ;;
+            \?)
+                err "illegal option -- $OPTARG"
+                usage
+                exit 1
+                ;;
+        esac
+    done
+
+    validate_opts
+    set_defaults
+}
+
+function validate_opts {
+    if test -n "$DISABLE_DB" -a -n "$DISABLE_MIGRATIONS"
+    then
+        err "cannot use -S and -M options together"
+        exit 2
+    fi
+
+    if test -n "$DISABLE_DB" -a -n "$SNAPSHOT_NAME"
+    then
+        err "cannot use -S and -s options together"
+        exit 2
+    fi
+
+    if test -n "$DISABLE_UR" -a -n "$UR_REPO"
+    then
+        err "cannot use -U and -u options together"
+        exit 2
+    fi
+
+    if test -n "$DISABLE_WF" -a -n "$WF_REPO"
+    then
+        err "cannot use -W and -w options together"
+        exit 2
+    fi
+
+    if test -n "$SNAPSHOT_NAME"
+    then
+        if ! test-db template list | cut -f 1 | grep -q "^${SNAPSHOT_NAME}$"
+        then
+            err "invalid snapshot name -- $SNAPSHOT_NAME"
+            exit 2
+        fi
+    fi
+
+    if test -n "$UR_REPO"
+    then
+        assert_dir "-u" "$UR_REPO"
+    fi
+
+    if test -n "$WF_REPO"
+    then
+        assert_dir "-w" "$WF_REPO"
+    fi
+}
+
+function assert_dir {
+    if ! test -d "$2"
+    then
+        err "$1 value must be a directory -- $2"
+        exit 2
+    fi
+}
+
+function set_defaults {
+    local TOP="$(git rev-parse --show-toplevel 2> /dev/null)"
+    local SUBMODULES
+
+    SUBMODULES=(gms-config configuration-manager)
+
+    if test -z "$SNAPSHOT_NAME"
+    then
+        SNAPSHOT_NAME="82f3e4d"
+    fi
+
+    if test -z "$UR_REPO"
+    then
+        if test -z "$TOP" -o ! -d "$TOP/ur"
+        then
+            err "-u must be specified if not in a Genome repo"
+            exit 2
+        fi
+        SUBMODULES=("${SUBMODULES[@]}" ur)
+        UR_REPO="$TOP/ur"
+    fi
+
+    if test -z "$WF_REPO"
+    then
+        if test -z "$TOP" -o ! -d "$TOP/workflow"
+        then
+            err "-w must be specified if not in a Genome repo"
+            exit 2
+        fi
+        SUBMODULES=("${SUBMODULES[@]}" workflow)
+        WF_REPO="$TOP/workflow"
+    fi
+
+    if test -n "$SUBMODULES"
+    then
+        prepare_submodules "${SUBMODULES[@]}"
+    fi
+}
+
+function __quiet_git {
+    (
+        set +o errexit
+        OUT="$(git "$@" 2>&1)"
+        GIT_EXIT=$?
+        if test $GIT_EXIT -ne 0
+        then
+            echo -e "$OUT" 1>&2
```

Comment:
Why is __quiet_git making so much noise? Is the idea to keep it quiet on stdout?